src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3687,6 +3700,37 @@ private bool AreEquivalentClosureScopes(SyntaxNode oldScopeOpt, SyntaxNode newSc
 
         #endregion
 
+        #region State Machines
+
+        private void ReportStateMachineRudeEdits(
+            UpdatedMemberInfo updatedInfo, 
+            ISymbol oldMember, 
+            List<RudeEditDiagnostic> diagnostics)
+        {
+            if (!updatedInfo.OldHasStateMachineSuspensionPoint)
+            {
+                return;
+            }
+
+            // only methods and anonymous functions may be async/iterators machines:
+            var stateMachineAttributeQualifiedName = ((IMethodSymbol)oldMember).IsAsync ?
+                "System.Runtime.CompilerServices.AsyncStateMachineAttribute" :
```

Comment:
Use AttributeDescription.AsyncStateMachineAttribute to get this name. Same for IteratorStateMachineAttribute